lib/Sema/CSSimplify.cpp

```diff
@@ -760,39 +760,6 @@ static ConstraintSystem::TypeMatchResult
 matchCallArguments(ConstraintSystem &cs, ConstraintKind kind,
                    Type argType, Type paramType,
                    ConstraintLocatorBuilder locator) {
-
-  if (paramType->isAny()) {
-    if (argType->is<InOutType>())
-      return cs.getTypeMatchFailure(locator);
-
-    // If the param type is Any, the function can only have one argument.
-    // Check if exactly one argument was passed to this function, otherwise
-    // we obviously have a mismatch.
-    if (auto tupleArgType = dyn_cast<TupleType>(argType.getPointer())) {
-      if (tupleArgType->getNumElements() != 1 ||
-          tupleArgType->getElement(0).hasName()) {
-        return cs.getTypeMatchFailure(locator);
-      }
-    }
-
-    // Disallow assignment of noescape function to parameter of type
-    // Any. Allowing this would allow these functions to escape.
-    if (auto *fnTy = argType->getAs<AnyFunctionType>()) {
```

Comment:
As I understand this is covering following code:

```swift
func foo(_: Any) {}
func bar(_ fn: (Int, Int) -> Void) {
  foo(fn) // error: converting non-escaping value to 'Any' may allow it to escape
}
```

If we remove it how is situation like that going to be diagnosed?

Reply:
We already diagnose this elsewhere, because we have to in the multi-parameter case.

Reply:
Maybe `matchTypes` handles it then, hope we have or will have some more tests for this, that's where another use of `MarkExplicitlyEscaping` is...